### DISCUSSION TITLE
No module named binascii

### DISCUSSION THREAD

**Discussion by 0-wiz-0** [pypa/setuptools#3021] [Q&A]
This might sound contrived but happened in practice (https://mail-index.netbsd.org/pkgsrc-users/2022/01/10/msg035036.html):
* install python 3.9.8
* install setuptools
* delete python 3.9.8 to make space for building and installing python 3.9.9
* build python 3.9.9
Now building python fails because python's setup.py finds setuptool's distutils which tries to use binascii, which is not installed yet/any longer.
Any ideas how to work around this problem?
Thanks.

**Comment by gdt** [comment]
Actually I did not delete the previous python 3.9.x.    I simply tried to build the new one.

It seems to me that when running the in-tree python that was just built to do things, it should be prevented from loading anything from the normal PYTHONPATH.

Deleting the installation of setuptools allowed the build to work, but that's of course a workaround.

(This is a clarification not an answer.)
